awx/main/fields.py

```diff
@@ -987,3 +990,108 @@ def from_db_value(self, value, expression, connection, context):
         if value and value.startswith('$encrypted$'):
             return decrypt_value(get_encryption_key('value', pk=None), value)
         return value
+
+
+class OrderedManyToManyDescriptor(ManyToManyDescriptor):
+    """
+    Django doesn't seem to support:
+
+    class Meta:
+        ordering = [...]
+
+    ...on custom through= relations for ManyToMany fields.
+
+    Meaning, queries made _through_ the intermediary table will _not_ apply an
+    ORDER_BY clause based on the `Meta.ordering` of the intermediary M2M class
+    (which is the behavior we want for "ordered" many to many relations):
+
+    https://github.com/django/django/blob/stable/1.11.x/django/db/models/fields/related_descriptors.py#L593
+
+    This descriptor automatically sorts all queries through this relation
+    using the `position` column on the M2M table.
+    """
+
+    @cached_property
+    def related_manager_cls(self):
+        model = self.rel.related_model if self.reverse else self.rel.model
+
+        def add_custom_queryset_to_many_related_manager(many_related_manage_cls):
+            class OrderedManyRelatedManager(many_related_manage_cls):
+                def get_queryset(self):
+                    return super(OrderedManyRelatedManager, self).get_queryset().order_by(
```

Comment:
It looks like the `django-sortedm2m` plugin does something _very_ similar to this:

https://github.com/gregmuellegger/django-sortedm2m/blob/master/sortedm2m/fields.py#L42

Reply:
@AlanCoding @jbradberry this seems like reasonable behavior for *Django* to have.  I might put together a PR upstream and see if there's any interest (it would be an easy PR).

Maybe we could land this upstream and get rid of this custom descriptor in the future.